src/app/pages/things/channels/channels.component.ts

```diff
@@ -162,4 +133,43 @@ export class ChannelsComponent implements OnInit {
       },
     );
   }
+
+  onFileSelected(files: FileList) {
+    if (files && files.length > 0) {
+      const file: File = files.item(0);
+      const reader: FileReader = new FileReader();
+      reader.readAsText(file);
+      reader.onload = () => {
+        const csv: string = reader.result as string;
+        const lines = csv.split('\n');
+        const channels: Channel[] = [];
+
+        lines.forEach( line => {
+          const col = line.split('|');
+          if (col[0] !== '' && col[0] !== '<empty string>') {
+            let metadata = {};
+            if (col[1] !== undefined) {
+              try {
+                metadata = JSON.parse(col[1]);
+              } catch (e) {
+                this.notificationsService.warn('Wrong metadata format', '');
+              }
+            }
+
```

Comment:
Maybe you coud assign `col[0]` and `col[1]` to meaningfully named consts like `name` and `metadata`.

Reply:
I mean, at the very top, so the meaning of code is clear during the `col[]` manipulation.